js/core.js

```diff
@@ -168,15 +168,24 @@
         /**
          * should be used to log information on the server to different log locations
          * @param {Object} headers - the headers to be logged, should include action
-         * @param {String} location - the path for logging (terminal_error || button_action)
+         * @param {String} location - the path for logging (terminal_error || client_action)
          **/
-        this.logHeaders = function (actionHeader, location) {
+        this.logHeaders = function (contextHeaderParams, location) {
             var defer = module._q.defer();
 
+            if (!contextHeaderParams || (contextHeaderParams === Object(contextHeaderParams) && !Array.isArray(contextHeaderParams))) {
```

Comment:
I think throwing error makes sense since the whole point of this function is to log the passed object.